Develop

Kanezoor opened the pull request:
https://kanezoor.github.io/js_2048_game/

Kanezoor committed:
saved before changes for ES modules

Kanezoor committed:
commit after the switching to ES modules

Kanezoor committed:
finished logically but failing tests

Kanezoor committed:
finished, tests done

vitalii-bondarenko-ua reviewed (changes requested):
Please, provide demo link to PR description.

Kanezoor commented:
https://kanezoor.github.io/js_2048_game/ ,sorry forgot about it completely.

vitalii-bondarenko-ua commented on src/modules/Game.class.js:
```diff
@@ -21,24 +21,126 @@ class Game {
+    this.state = this._initialState.map((row) => [...row]);
+
+    // // eslint-disable-next-line no-console
+    // console.log(initialState);
```
remove useless commented code

vitalii-bondarenko-ua reviewed (changes requested):
Good job.
Needs some improvements.
Check comment.
And such behavior: after click to restart button should be visible start message and button should have start textContent and styles.

https://github.com/user-attachments/assets/8cc604e4-9de8-4b60-874f-3fca8535eb28

Kanezoor committed:
cleaned all unnecessary comments

Kanezoor committed:
changed behaviour of start message

Kanezoor commented:
hope I addressed all the issues from your review

vitalii-bondarenko-ua reviewed (approved):
Well done

Kanezoor committed:
added preview picture

Kanezoor committed:
Revise README for improved clarity and structure

Updated README to enhance clarity and structure, added sections for key features, technologies used, and instructions for running the game locally.
